processor/filterprocessor/README.md

```diff
@@ -76,15 +80,29 @@ processors:
           - Key: host.name
             Value: just_this_one_hostname
     logs/regexp:
+      include:
         match_type: regexp
         resource_attributes:
           - Key: host.name
             Value: prefix.*
     logs/regexp_record:
+      include:
         match_type: regexp
         record_attributes:
           - Key: record_attr
             Value: prefix_.*
+    logs/severity:
+      include:
+        match_type: regexp
+        severity_texts:
+        - INFO[2-4]?
+        - WARN[2-4]?
+        - ERROR[2-4]?
```

Comment:
Should these be made case insensitive? 

Technically, severity text is not constrained to match the severity level, so defining it narrowly like this might cause some unexpected problems. 

Really, filtering on severity itself would be more useful, but I think the point here is to match existing functionality from `attributesprocessor`.

Reply:
I think these should be case sensitive; This is the current behavior consistent with all the fields if I understand correctly, and also shared by the attributes processor (they use the same underlying internal library with this PR).

I opened this issue https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/12959 and think it would be a good follow-up to allow filtering based on the SeverityNumber (which is more constrained and standard)

Reply:
Most commonly, severity text will follow this naming style, so I can accept that the documentation should show this.

To be clear though, severity text is not rigidly defined as these particular values. Any string value may be present. Many of the collector's log receivers can be configured to interpret any value as a severity, but [the severity text will be preserved as the original value](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/c7486b476f6a51eb5c81c72041929ed353627fbd/pkg/stanza/operator/helper/severity.go#L57). (eg. You can easily end up with: `severity: 17` and `severity_text: "bad"`)